bazel/sh_test_wrapper.sh

```diff
@@ -3,4 +3,6 @@
 # Where the runfiles are for tests.
 export TEST_RUNDIR="${TEST_SRCDIR}/${TEST_WORKSPACE}"
 
+cd $(dirname "$0")
```

Comment:
So, cc_configure.bzl defines a `repository_rule`, which gets a `repository_ctx` object that contains the os information.

`sh_test` (which is the rule `envoy_sh_test` eventually creates) is a regular rule. Regular rules get a `ctx` object, which does not contain os information. In any event, it's not really OS that we want to key on but whether or not hot restart has been disabled for the build.

The `ctx` object does have a `vars` dict which contains "hot_restart": "disabled" when building hot restart is disabled, but entry that won't exist on OS X, since I'm trying to disable hot restart there by default (as it won't even compile). I suppose we could say that building on OS X requires explicitly disabling the feature with a command line flag.

At that point, I'm left writing a rule that duplicates the functionality of `sh_test` with the features available in `ctx`. And at that point I'm stuck. I can get it to run the wrapper script and am left concatenating strings to try to reproduce the TEST_XXX environment variables that the test depends on. In particular, a value for TEST_TMPDIR just doesn't seem to be available.

In the end, even if I got that to work, it's not at all clear to me that it's better.